src/main.ts

```diff
@@ -563,6 +574,15 @@ async function getHashForBlock(
 
 		return await api.rpc.chain.getBlockHash(blockNumber);
 	} catch (err) {
+		// Check if the block number is too high
+		const { number } = await api.rpc.chain.getHeader();
+		if (blockNumber && number.toNumber() < blockNumber) {
+			throw new BadRequest(
```

Comment:
This is a breaking change because we are now returning a different HTTP response code so I would suggest keeping this until v1.

Reply:
I see the argument that it is a breaking change since we introduce a new error that potentially prevents an older error message from appearing. However I think there are two reasons this is ok right now. 1) we have a precedent for introducing these types of error handling changes pre v1 (see #123 as a recent example) and 2) it is replacing downstream error messages that I see as significantly less helpful.

Reply:
It's not the error _message_ that is really the problem, it's the error _code_. This could break integrations if people are expecting a `5xx` type error when the block number isn't right.

Reply:
I see what you mean about #123 though...

Reply:
What do you think of changing it to a 500 but leaving the rest?

Reply:
I think our user group is small enough that we can ask them...

Reply:
If we _hadn't_ have already released #123 I would have gone with Zeke's suggestion but given the state of things, I think Joe's is more reasonable.